fuel-vm/src/checked_transaction.rs

```diff
@@ -332,11 +334,54 @@ pub trait IntoChecked: FormatValidityChecks + Sized {
     ) -> Result<Checked<Self>, CheckError>
     where
         Checked<Self>: CheckPredicates,
+    {
+        self.into_checked_reusable_memory_ecal(
+            block_height,
+            consensus_params,
+            memory,
+            storage,
+            NotSupportedEcal,
+        )
+    }
+
+    /// Returns transaction that passed all `Checks`.
+    fn into_checked_ecal<Ecal>(
```

Comment:
nit: Do we want these `_ecal` methods to be on a separate ext trait? for both the `checked` and `estimate` traits? Maybe it doesn't matter.

Reply:
It was just a subtle interface segregation example, since the `__ecal` methods won't be used by the same consumers. i.e. it is coupling the `checked` and `estimate` interfaces with debug features.

It's like how you don't want to include test code in production. I wouldn't want to include test methods on a trait without a `#[cfg(test)]` or `#[cfg(feature=test-helpers)]`. Since this is a runtime feature we don't have that luxary. 

Again this is a "nit", so I'm not going to push too hard on this.

> I'd rather consider just removing the non _ecal methods (and not appending _ecal from the name).

Yeah. Good point. And at that point we might as well have a default `impl` for the `NotSupportedEcal`. 

I think this is noop.